vulture/config.py

```diff
@@ -0,0 +1,184 @@
+"""
+This module handles retrieval of configuration values from either the
+command-line arguments or the pyproject.toml file.
+"""
+from __future__ import print_function
+import argparse
+import sys
+from os.path import abspath, exists
+from typing import Any, Dict
+
+import toml
+
+from .version import __version__
+
+
+class Config(Dict[str, Any]):
+    """
+    A config object wrapping various variables and used to abstract away the
+    difference between CLI-arg parsing and TOML loading.
+    """
+
+    #: Possible configuration options and their respective defaults
+    DEFAULTS = {
+        "min_confidence": 0,
+        "paths": [],
+        "exclude": [],
+        "ignore_decorators": [],
+        "ignore_names": [],
+        "make_whitelist": False,
+        "sort_by_size": False,
+        "verbose": False,
+    }
+
+    def __getattribute__(self, name):
+        if name in self:
+            return self[name]
+        return super(Config, self).__getattribute__(name)
```

Comment:
I wonder whether the code wouldn't be simpler if we didn't subclass dict, but used dict directly? What do you think? I think the only thing to change would be to turn the staticmethod into a function that returns a dict.

Reply:
Addressed this in 06ebedd

Reply:
I agree about this. As mentioned I felt iffy about this as well. And additionally the overriding of `__getattribute__` would break things like `foo.keys()`.

Originally I added this as a poor-man's implementation of a data-class for type-safety, but as typing is not yet in vulture, and because we need to support Python <= 3.5 this is a no-go.

Removed the subclass :)